p2p/sql_stmt.go

```diff
@@ -336,12 +487,22 @@ func (srv *Server) addNodeMetaInfo(nodeid string, hash string, dial bool, accept
 		log.Crit("No prepared statement for AddNodeMetaInfo")
 		return
 	}
-
-	_, err := srv.addNodeMetaInfoStmt.Exec(nodeid, hash, boolToInt(dial), boolToInt(accept), boolToInt(tooManyPeers))
+	p2pDisc4, ethDisc4 := false, false
+	if tooManyPeers {
+		if dial || accept {
```

Comment:
I don't quite follow the logic here. Why is ethDisc being set to true if `!dial && !accept`?

Reply:
`NodeMetaInfo` is updated under following 3 conditions:
1. successful connection: increment `dial_count` or `accept_count` (`p2p/node_info.go:242`)
2. connection fail due to `TooManyPeers` during p2p handshake: increment `dial_count` or `accept_count`, increment `p2p_disc4_count` (`p2p/server.go:807`)
3. connection fail due to `TooManyPeers` during eth handshake: increment `eth_disc4_count` (not included in this PR yet).
Case 3 is not implemented in this PR because I was trying to avoid merge conflict (I think..), but that is what the `ethDisc = !dial && !accept` logic is for. It is to avoid incrementing `dial_count` and `accept_count` when `TooManyPeers` disconnect message is received during Ethereum protocol handshake (after DevP2P connection is established).
The function call for case 3 looks like this: `srv.addNodeMetaInfo(id.String(), nodeInfo.Keccak256Hash, dial: false, accept: false, true)` where both `dial` and `accept` are explicitly set as false.

Reply:
👍